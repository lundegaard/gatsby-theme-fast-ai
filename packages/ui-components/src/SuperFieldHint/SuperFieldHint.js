import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const SuperFieldHint = forwardRef(({ children, sx, ...rest }, ref) => (
	<Text
		ref={ref}
		fontSize={[0, 1]}
		mb={0}
		sx={{ minHeight: '16px', fontSize: 0, letterSpacing: 'mono', ...sx }}
		{...rest}
	>
		{children}
	</Text>
));

SuperFieldHint.propTypes = {
	children: PropTypes.node,
	sx: PropTypes.object,
};

SuperFieldHint.displayName = 'SuperFieldHint';

export default SuperFieldHint;
