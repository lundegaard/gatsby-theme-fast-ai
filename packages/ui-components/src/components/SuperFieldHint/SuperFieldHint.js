import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const SuperFieldHint = ({ children, sx, ...rest }) => (
	<Text
		fontSize={[0, 1]}
		sx={{ minHeight: '16px', fontSize: 0, letterSpacing: 'mono', ...sx }}
		{...rest}
	>
		{children}
	</Text>
);

SuperFieldHint.propTypes = {
	children: PropTypes.node,
	sx: PropTypes.object,
};

export default SuperFieldHint;
