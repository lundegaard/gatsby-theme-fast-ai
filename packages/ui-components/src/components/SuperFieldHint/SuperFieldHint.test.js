import React from 'react';
import { shallow } from 'enzyme';

import SuperFieldHint from './SuperFieldHint';

describe('<SuperFieldHint />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<SuperFieldHint {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
