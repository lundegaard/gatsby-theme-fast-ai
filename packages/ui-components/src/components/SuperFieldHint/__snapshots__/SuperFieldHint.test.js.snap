// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuperFieldHint /> matches snapshot when shallowly rendered 1`] = `
<Text
  fontSize={
    Array [
      0,
      1,
    ]
  }
  mb={0}
  sx={
    Object {
      "fontSize": 0,
      "letterSpacing": "mono",
      "minHeight": "16px",
    }
  }
/>
`;
